misc/scripts/check_mixed_eol.sh

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+
+cd `dirname "$0"`/../..
+
+exitcode=0
+
+check_eol()
+{
+    if [[ $1 == *.ani  || \
+          $1 == *.bmp  || \
+          $1 == *.chm  || \
+          $1 == *.dia  || \
+          $1 == *.gif  || \
+          $1 == *.gz   || \
+          $1 == *.hlp  || \
+          $1 == *.icns || \
+          $1 == *.ico  || \
+          $1 == *.jpg  || \
+          $1 == *.mo   || \
+          $1 == *.mpg  || \
+          $1 == *.pcx  || \
+          $1 == *.pdf  || \
+          $1 == *.png  || \
+          $1 == *.pnm  || \
+          $1 == *.tga  || \
+          $1 == *.tif  || \
+          $1 == *.ttf  || \
+          $1 == *.wav  || \
+          $1 == *.zip     \
+       ]]; then
+        return;
+    fi
+
+    read dos unix mac <<< $(dos2unix --info=dum "$1" | awk '{print $1, $2, $3}')
```

Comment:
I don't know if using `dos2unix` is better or worse but, just in case you find it useful, here is my own `eoltype` script I use for EOL detection and which is battle-tested since a few years:
```perl
#!/usr/bin/env perl
use strict;
use warnings;
use v5.10;

use autodie;

binmode STDIN, ':raw';

my $crlf = -1;
my $mixed = 0;
while (<>) {
    # We are not interested in checking binary files.
    if (/\x00/) {
        say "binary";
        exit 0
    }

    # Ignore the last line without any EOL at all.
    last unless /\n$/;

    my $line_crlf = /\r\n$/;
    if ($crlf == -1) {
        $crlf = $line_crlf;
    } elsif ($line_crlf != $crlf) {
        # Note that we need to continue in case this is a binary file.
        $mixed=1
    }
}

if ($mixed) {
    say "mixed";
    exit 1
}

if ($crlf == -1) {
    say "none"
} elsif ($crlf == 0) {
    say "unix"
} elsif ($crlf == 1) {
    say "dos"
}

exit 0
```

Reply:
`dos2unix` returns a simple summary of the used line endings, but similarly I have to skip binary files.
```
     DOS    UNIX     MAC  FILE
       0      37       0  check_mixed_eol.sh
```